crates/wasm/src/wasm_planner.rs

```diff
@@ -0,0 +1,225 @@
+}
+
+#[wasm_bindgen]
+impl WasmPlanner {
+    #[wasm_bindgen(constructor)]
+    pub fn new() -> WasmPlanner {
+        WasmPlanner {
+            planner: Planner::new(OsRng),
+        }
+    }
+
+    pub fn expiry_height(&mut self, expiry_height: JsValue) -> Result<(), JsValue> {
+        self.planner
+            .expiry_height(serde_wasm_bindgen::from_value(expiry_height)?);
+
+        Ok(())
+    }
+
+    pub fn memo(&mut self, memo: JsValue) -> Result<(), JsValue> {
+        let memo_proto: MemoPlaintext = serde_wasm_bindgen::from_value(memo)?;
+
+        let _ = self.planner.memo(memo_proto.try_into().unwrap());
+
+        Ok(())
+    }
+
+    pub fn fee(&mut self, fee: JsValue) -> Result<(), JsValue> {
+        let fee_proto: Fee = serde_wasm_bindgen::from_value(fee)?;
+
+        self.planner.fee(fee_proto.try_into().unwrap());
+
+        Ok(())
+    }
+
+    pub fn output(&mut self, value: JsValue, address: JsValue) -> Result<(), JsValue> {
+        let value_proto: Value = serde_wasm_bindgen::from_value(value)?;
+        let address_proto: Address = serde_wasm_bindgen::from_value(address)?;
+
+        self.planner.output(
+            value_proto.try_into().unwrap(),
+            address_proto.try_into().unwrap(),
+        );
+
+        Ok(())
+    }
+
+    pub async fn swap_claim(&mut self, swap_commitment: JsValue) -> Result<(), JsValue> {
+        utils::set_panic_hook();
+
+        let swap_commitment_proto: StateCommitment =
+            serde_wasm_bindgen::from_value(swap_commitment)?;
+
+        let swap_record = get_swap_by_commitment(swap_commitment_proto).await.unwrap();
+        let chain_params_proto: ChainParameters = get_chain_parameters().await.unwrap();
+
+        let swap_claim_plan = SwapClaimPlan {
+            swap_plaintext: swap_record.swap,
+            position: swap_record.position,
+            output_data: swap_record.output_data,
+            epoch_duration: chain_params_proto.epoch_duration,
+            proof_blinding_r: Fq::rand(&mut OsRng),
+            proof_blinding_s: Fq::rand(&mut OsRng),
+        };
+
+        self.planner.swap_claim(swap_claim_plan);
+        Ok(())
+    }
+
+    pub fn swap(
+        &mut self,
+        input_value: JsValue,
+        into_denom: JsValue,
+        swap_claim_fee: JsValue,
+        claim_address: JsValue,
+    ) -> Result<(), JsValue> {
+        let input_value_proto: Value = serde_wasm_bindgen::from_value(input_value)?;
+        let into_denom_proto: DenomMetadata = serde_wasm_bindgen::from_value(into_denom)?;
+        let swap_claim_fee_proto: Fee = serde_wasm_bindgen::from_value(swap_claim_fee)?;
+        let claim_address_proto: Address = serde_wasm_bindgen::from_value(claim_address)?;
+
+        let _ = self.planner.swap(
+            input_value_proto.try_into().unwrap(),
+            into_denom_proto.try_into().unwrap(),
+            swap_claim_fee_proto.try_into().unwrap(),
+            claim_address_proto.try_into().unwrap(),
+        );
+
+        Ok(())
```

Comment:
What's the purpose of this function?

Reply:
This is part of the functionality for flexible TransactionPlan creation
An example of use can be found [here](https://github.com/penumbra-zone/wallet/blob/dev/src/services/ViewProtocolService.ts#L279C4-L279C4)
With this function we add Swap actions to planner